pkg/pillar/cmd/vcomlink/vcomlink.go

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2018-2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package vcomlink
+
+import (
+	"encoding/json"
+	"flag"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/agentbase"
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/pubsub"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/vcom"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	agentName     = "vcomlink"
+	errorTime     = 3 * time.Minute
+	warningTime   = 40 * time.Second
+	maxPacketSize = 4096
+	backLogSize   = unix.SOMAXCONN
+)
+
+var (
+	logger *logrus.Logger
+	log    *base.LogObject
+)
+
+type vcomLinkContext struct {
+	agentbase.AgentBase
+	ps              *pubsub.PubSub
+	subGlobalConfig pubsub.Subscription
+
+	GCInitialized bool
+	// cli options
+}
+
+// AddAgentSpecificCLIFlags adds CLI options
+func (ctx *vcomLinkContext) AddAgentSpecificCLIFlags(flagSet *flag.FlagSet) {
+}
+
+// Run is the entry point for vcomlink, from zedbox
+//
+//nolint:funlen,gocognit,gocyclo
```

Comment:
I don't think we need those supressions.

Reply:
removed.